src/cascadia/TerminalControl/ControlInteractivity.cpp

```diff
@@ -182,7 +182,12 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         }
         else if (_canSendVTMouseInput(modifiers))
         {
-            _core->SendMouseEvent(terminalPosition, pointerUpdateKind, modifiers, 0, buttonState);
+            auto adjustment = 0;
+            if (_core->ScrollOffset() > 0)
+            {
+                adjustment = _core->BufferHeight() - _core->ScrollOffset() - _core->ViewHeight();
+            }
+            _core->SendMouseEvent({ terminalPosition.x(), terminalPosition.y() - adjustment }, pointerUpdateKind, modifiers, 0, buttonState);

```

Comment:
Primary concern:

Can someone think of how this may unintentionally affect/break other things?

Reply:
So, this is interesting. Inside SendMouseEvent, it locks the Y coordinate to 0. Perhaps this is correct...

It would be interesting to see what other terminal emulators do in raw mouse mode when they are scrolled up.

Reply:
It looks like gnome-terminal, at least, completely ignores the event. It just blackholes it.

Reply:
If you want, you could future-proof this for [Infinite Scrollback](https://github.com/microsoft/terminal/issues/1410) by `std::max(terminal.Position.y() - adjustment, 0)`? That's the only thing I can think of.

Reply:
We follow gnome terminal! Clicks outside the active region are ignored